app-creator/client/src/widgets/dropzone-widget/dropzone-widget.ts

```diff
@@ -130,36 +124,45 @@ export class Dropzone extends LitElement {
    */
   handleAddingWidget(
     parent: Element | null,
-    container: Element,
     widget: Element,
     nextElement: Element | null
   ) {
     // Making clone with new id.
     const clone = widget.cloneNode(true) as HTMLElement;
     clone.id += `-${store.getState().widgetIDs[widget.id]}`;
 
+    // Return early if the widget has been added to another panel earlier.
+    const template = store.getState().template;
+    for (const id in template) {
+      if (
+        typeof template[id] === 'object' &&
+        !Array.isArray(template[id]) &&
+        id !== parent!.id &&
+        template[id].children.includes(clone.id)
+      ) {
```

Comment:
One more question about this check: does this mean a widget cannot be "copied" to a new panel if it's already in another panel?

Reply:
Widgets can be moved between panels freely. This case is when we are dragging a widget for the first time from the widgets tab. We are restricting adding to one panel initially because we increment the widget IDs on dragend events. This way we can know for sure that we added only one element at each drag start and drag end cycle. It fixes an issue with inconsistent IDs.

Reply:
Got it, thanks!